src/catfs/rlibc.rs

```diff
@@ -296,7 +322,10 @@ pub fn fstatat(dir: RawFd, path: &AsRef<Path>) -> io::Result<libc::stat> {
     let stp = (&mut st) as *mut libc::stat;
     let s = to_cstring(path);
 
+    #[cfg(not(target_os = "macos"))]
     let res = unsafe { libc::fstatat(dir, s.as_ptr(), stp, libc::AT_EMPTY_PATH) };
+    #[cfg(target_os = "macos")]
+    let res = unsafe { libc::fstatat(dir, s.as_ptr(), stp, 0) };
```

Comment:
Nope, this won't work for an empty path, which is exactly what `AT_EMPTY_PATH` is meant to do. Just run this C code on a Mac:

```C
#include <stdio.h>
#include <sys/stat.h>
#include <fcntl.h>
int main(void) {
    struct stat buf;
    printf("%d\n", fstatat(AT_FDCWD, "", &buf, 0));
    perror("fstatat");
    return 0;
}
```

A racy substitute might be needed, using a path from [`F_GETPATH` fcntl](https://stackoverflow.com/a/13544447/3770260).